lib/genesis_collector/megaraid.rb

```diff
@@ -0,0 +1,52 @@
+module GenesisCollector
+  module MegaRaid
+
+    def collect_megaraid
+      @payload[:raids] ||= []
+      @payload[:properties] ||= {}
+      if megaraid?
+        @payload[:properties]['MEGARAID_TYPE'] = read_megacli_adapter_type
+        @payload[:properties]['MEGARAID_FW_PACKAGE'] = read_megacli_adapter_firmware
+        @payload[:raids].concat read_megacli_logical_disks
+      end
+    end
+
+    def megaraid?
+      return !read_megacli_adapter_type.nil?
```

Comment:
Ruby automatically returns the last line in a function so you don't need the `return` keyword here.